File: 2309 - 일곱  난쟁이/2309.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#define endl '\n'

using namespace std;

int main(){
    vector <int> v1;
    vector <int> v2;
    int sum = -100;
    int a, b, flag = 0;

    for (int i = 0; i < 9; i++){
        int tmp;
        cin >> tmp;
        sum += tmp;
        v1.push_back(tmp);
    }
    // 가짜 a와 b 찾기
    for (int i = 0; i < 8; i++){
        for (int j = 1; j < 9 - i; j++){
            if (sum == v1[i] + v1[i + j]){
                a = v1[i];
                b = v1[i + j];
                flag = 1;
                break;
            }
        }
        if (flag == 1) break;
    }

    // a와 b를 제외하고 오름차순으로 출력하기
    for (int i = 0; i < 9; i++){
        if (v1[i] != a && v1[i] != b){
            v2.push_back(v1[i]);
        }  
    }
    sort(v2.begin(), v2.end());
    for (int i = 0; i < 7; i++) cout << v2[i] << endl;
}

File: 2309 - 일곱  난쟁이/2309.py
import sys

# 입력
LST = []
for i in range(9):
    n = int(sys.stdin.readline())
    LST.append(n)

# 두 가짜 찾기
N = sum(LST) - 100 # 40
a = b = 0
flag = 0
for i in range(8):
    for j in range(1, 9 - i):
        if N == LST[i] + LST[i + j]:
            a, b = LST[i], LST[i + j]
            flag = 1
            break
    if flag == 1: break

# 가짜 없애고 출력하기
LST.remove(a)
LST.remove(b)
LST.sort()
for i in LST: print(i)
